File: suraj_24b0907/week7/climbingstairs.cpp
#include <bits/stdc++.h>
using namespace std;
class Solution {
public:
    vector<int> v={0,1,2,3};
    int climbStairs(int n) {
        if(n<v.size()) return v[n];
        while(v.size()<=n){
            v.push_back(v[v.size() - 1] + v[v.size() - 2]);
        }
        return v[n];
    }
};

File: Chavda_23B0368/Week_7/Climbing_stairs.java
package Week_7;

class Solution {
    public int climbStairs(int n) {
        if(n == 1 || n == 2)
            return n;
        if(n <= 0)
            return 0;
        int prev2 = 1;
        int prev = 2;
        int curr = 0;
        for(int i = 3; i <= n; i++) {
            curr = prev + prev2;
            prev2 = prev;
            prev = curr;
        }
        return curr;
    }
}